resource/modules/resource_match.cpp

```diff
@@ -2042,6 +2083,49 @@ static void cancel_request_cb (flux_t *h, flux_msg_handler_t *w,
         flux_log_error (h, "%s: flux_respond_error", __FUNCTION__);
 }
 
+static void partial_cancel_request_cb (flux_t *h, flux_msg_handler_t *w,
+                                       const flux_msg_t *msg, void *arg)
+{
+    std::shared_ptr<resource_ctx_t> ctx = getctx ((flux_t *)arg);
+    int64_t jobid = -1;
+    char *R = NULL;
+    std::map<uint64_t, uint64_t>::iterator jobid_it;
```

Comment:
If we can't use auto for things like iterators because of placement and gotos for now, perhaps we can at least use decltype on the variable so if we decide to change the specific type later it's less painful.

```suggestion
    decltype(ctx->allocations)::iterator jobid_it;
```

Reply:
That's slick; thanks! Implemented.